File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/Profile/AdminProfileDTO.cs
using platform_core_service.Common.Entities.BaseEntity;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.Profile
{
    public class AdminProfileDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;
        public string UserId { get; set; } = null!;
        public string DisplayName { get; set; } = null!;
        public string? Role { get; set; }
        public bool IsSuspended { get; set; }
        public DateTimeOffset? SuspendedUntil { get; set; }
        public DateTimeOffset? CreatedAt { get; set; }
        public int PostCount { get; set; }
    }
}

File: frontend/src/types/admin/admin-profile-dto.ts
export interface AdminProfileDTO {
  id: string;
  userId: string;
  displayName: string;
  email?: string | null;
  avatarUrl?: string | null;
  reputationPoints: number;
  role?: string | null;
  isSuspended: boolean;
  suspendedUntil?: string | null;
  createdAt?: string | null;
  postCount: number;
}

export interface AdminSuspendUserDTO {
  daySuspend?: number | null;
  reason?: string | null;
}

export interface AdminUpdateRoleDTO {
  newRole: string;
}
